script.js

```diff
@@ -1,4 +1,155 @@
+// Global variables
+var GAME_STATE_DICE_ROLL = "GAME_STATE_DICE_ROLL";
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var INSTRUCTIONS_FOR_NEXT_GAME_STATE = `Please choose the order of the dice by inputting 1 or 2`;
+
+var currentPlayerRolls = [];
+
+var player1ScoreCounter = 0;
+var player2ScoreCounter = 0;
+
+var currentPlayer = 1;
+var allPlayersScore = [];
+
+var mode = "";
+var regularMode = "regular";
+var lowestMode = "lowest";
+var MODE_SELECTION_INSTRUCTIONS = `please choose a mode to play:<br><br>${regularMode} or ${lowestMode}`;
+
+// Helper Function
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+var rollDiceForPlayer = function () {
+  var counter = 0;
+  for (var counter = 0; counter < 2; counter += 1) {
+    currentPlayerRolls.push(diceRoll());
+    console.log("error check " + currentPlayerRolls);
+  }
+  return `Player ${currentPlayer}, your rolls are ${currentPlayerRolls[0]} and ${currentPlayerRolls[1]}<br><br>${INSTRUCTIONS_FOR_NEXT_GAME_STATE}`;
+};
+
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  //playerInput == 1
+  if (playerInput == 1) {
+    playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+  //playerInput == 2
+  if (playerInput == 2) {
+    playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+
+  // Store playerScore in array
+  allPlayersScore.push(playerScore);
+
+  //clear current player rolls array
+  currentPlayerRolls = [];
+
+  return `Player ${currentPlayer}, your chosen value is: ${playerScore}`;
+};
+
+var comparePlayersScores = function () {
+  compareMessage = `Player 1 value: ${allPlayersScore[0]}<br>Player 2 value: ${allPlayersScore[1]}`;
+  var checkPlayer1Win = false;
+  if (allPlayersScore[0] == allPlayersScore[1]) {
+    compareMessage = `${compareMessage}<br><br>It's a tie!`;
+    return compareMessage;
+  }
+  if (allPlayersScore[0] > allPlayersScore[1]) {
+    checkPlayer1Win = true;
+  }
+  if (mode == lowestMode) {
+    checkPlayer1Win = !checkPlayer1Win;
+  }
+  if (checkPlayer1Win == true) {
```

Comment:
the `if(booleanVariable == true)` here is a bit redundant.

`booleanVariable` already has a true/false value, and while you can compare it with a static boolean value, it kinda already is one.

in this instance `if(checkPlayer1Win)` already suffice!